core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -3016,6 +3023,9 @@ static final class InFlightState {
         private InFlightState rollbackState;
         // The timer task for the acquisition lock timeout.
         private AcquisitionLockTimerTask acquisitionLockTimeoutTask;
+        // The boolean determines if the record has achieved a final state of ARCHIVED from which it cannot transition
+        // to any other state. This could happen because of LSO movement etc.
+        private boolean isMarkedArchived = false;
```

Comment:
If this is set to `true`, it means that the state of the record is ARCHIVED, and that there will be no further state transitions allowed? That seems like a terminal state to me, as opposed to "marked archived" which is a bit more nebulous in meaning.